tableauserverclient/models/property_decorators.py

```diff
@@ -46,3 +46,34 @@ def wrapper(self, value):
         return func(self, value)
 
     return wrapper
+
+
+def property_is_valid_time(func):
+    @wraps(func)
+    def wrapper(self, value):
+        units_of_time = {"hour", "minute", "second"}
+
+        if not any(hasattr(value, unit) for unit in units_of_time):
+            error = "Invalid time object defined."
+            raise ValueError(error)
+        return func(self, value)
+
+    return wrapper
+
+
+def property_is_int(min, max=2**31):
+    def property_type_decorator(func):
+        @wraps(func)
+        def wrapper(self, value):
+            if min is None:
+                return func(self, value)
```

Comment:
I could enable that, but then how would we say "this should be an int but not checked within a range"

`property_is_int(None)` is nice and clean to say that.
`property_is_int(None, 100)` is ambiguous with the above call because both min and max are required positional arguments, so no nice signature :(

I could:
1. Make a keyword argument that is `property_is_int(range=False)` or some similar keyword.